### ISSUE TITLE
Hook with GriefPrevention

### ISSUE THREAD

**Issue by Focxer** [MyPetORG/MyPet#1527]
When you call your pet inside a GriefPrevention claim, console goes crazy

```
[21:01:40 ERROR]: Could not pass event EntityDamageByEntityEvent to GriefPrevention v16.17.1-2b69191
java.lang.ClassCastException: class de.Keyle.MyPet.compat.v1_16_R3.entity.CraftMyPet cannot be cast to class org.bukkit.entity.Panda (de.Keyle.MyPet.compat.v1_16_R3.entity.CraftMyPet is in unnamed module of loader org.bukkit.plugin.java.PluginClassLoader @29a205bf; org.bukkit.entity.Panda is in unnamed module of loader 'app')
        at me.ryanhamshire.GriefPrevention.EntityEventHandler.isMonster(EntityEventHandler.java:667) ~[?:?]
        at me.ryanhamshire.GriefPrevention.EntityEventHandler.handleEntityDamageEvent(EntityEventHandler.java:714) ~[?:?]
        at me.ryanhamshire.GriefPrevention.EntityEventHandler.onEntityDamage(EntityEventHandler.java:696) ~[?:?]
        at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor803.execute(Unknown Source) ~[?:?]
        at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69) ~[patched_1.16.5.jar:git-Paper-794]
        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[patched_1.16.5.jar:git-Paper-794]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[patched_1.16.5.jar:git-Paper-794]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:624) ~[patched_1.16.5.jar:git-Paper-794]
        at org.bukkit.craftbukkit.v1_16_R3.event.CraftEventFactory.callEvent(CraftEventFactory.java:257) ~[patched_1.16.5.jar:git-Paper-794]
        at org.bukkit.craftbukkit.v1_16_R3.event.CraftEventFactory.callEntityDamageEvent(CraftEventFactory.java:1091) ~[patched_1.16.5.jar:git-Paper-794]
        at org.bukkit.craftbukkit.v1_16_R3.event.CraftEventFactory.handleEntityDamageEvent(CraftEventFactory.java:978) ~[patched_1.16.5.jar:git-Paper-794]
        at org.bukkit.craftbukkit.v1_16_R3.event.CraftEventFactory.handleEntityDamageEvent(CraftEventFactory.java:934) ~[patched_1.16.5.jar:git-Paper-794]
        at org.bukkit.craftbukkit.v1_16_R3.event.CraftEventFactory.handleLivingEntityDamageEvent(CraftEventFactory.java:1123) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.EntityLiving.damageEntity0(EntityLiving.java:1932) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.EntityLiving.damageEntity(EntityLiving.java:1260) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.MobEffectList.applyInstantEffect(MobEffectList.java:100) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.EntityAreaEffectCloud.tick(EntityAreaEffectCloud.java:339) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.WorldServer.entityJoinedWorld(WorldServer.java:960) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.World.a(World.java:955) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.WorldServer.doTick(WorldServer.java:646) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.MinecraftServer.b(MinecraftServer.java:1495) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.DedicatedServer.b(DedicatedServer.java:436) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.MinecraftServer.a(MinecraftServer.java:1347) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1135) ~[patched_1.16.5.jar:git-Paper-794]
        at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:291) ~[patched_1.16.5.jar:git-Paper-794]
        at java.lang.Thread.run(Thread.java:831) [?:?]
```

This server is running Paper version git-Paper-794 (MC: 1.16.5) (Implementing API version 1.16.5-R0.1-SNAPSHOT)

**Comment by Jakllp** [comment]
Oh boi I feared this would happen... RIP

**Comment by Focxer** [comment]
if you want, you could add an option to disable pets in claims for now, an option to despawn the pet after release him should exist too, cuz players can bring mobs to my Spawn and then release them there... i blocked pets on Spawn to solve this

btw, your plugin is awesome, i'm still using since 1.5.2 idk, myPet premium didnt exist yet, this year youre doing a good work on this plugin, no more bugs at least <3 THX so MUCH youre the best

**Comment by Jakllp** [comment]
Nah I think I'll have a fix ready tomorrow (my time so like... in ~12h)

And thanks for the kind words ^^

**Comment by Jakllp** [comment]
Fixed with `3.12-SNAPSHOT-B1678`
By https://github.com/MyPetORG/MyPet/commit/77abbd90ff2540d200ff486426303f0e2418db70
